lib/commands/push.ts

```diff
@@ -374,6 +374,21 @@ export default class PushCmd extends Command {
 			organization: [Organization];
 		};
 
+		// * Temporary fix *
+		// When doing `$expand organization` on a public app (when not the owner)
+		// a partial document is returned without any organization data.
+		// So e.g. `balena push balena-sound` will break this commands logic.
+		// The balena cloud builder api will soon change to accept application slugs
+		// instead of `owner, app`, after which we will not need to do the expand
+		//
+		// Users should not be pushing to public apps anyway, so for now catch this situation:
+		if (!application.organization[0]) {
+			throw new ExpectedError(stripIndent`
+			You do no have rights to push to the application ${application.app_name}.
+			If you are trying to deploy a public app, please make sure that you have created the application first.
```

Comment:
* Is this because only the owner of a public app can push to it? If so, perhaps it would be informative to users to state it:
```

	If you are trying to deploy a public app, please make sure that you are
	the application owner.
```

* NIT: Line exceeds 80 characters (no line break).

Reply:
> Is this because only the owner of a public app can push to it

That's my understanding yes.   

> If you are trying to deploy a public app, please make sure that you are the application owner.
	
I'm not sure.  This bug was discovered by a user that was trying to deploy a public app, but misunderstood the process.  Would that message help them?  They might read it as "you need to be the owner of the public app to deploy it".  Whereas of course we are talking about two "apps" (the public app, and the users app they will deploy it too).

Reply:
I see, I was misunderstanding it too. :-) Yes, the original version is better. Perhaps with a different minor twist: "your own application" instead of "the application" -
```
If you are trying to deploy a public app, please make sure that you have created
your own application first.
```